src/Illuminate/Database/Query/Grammars/PostgresGrammar.php

```diff
@@ -734,13 +736,15 @@ protected function wrapJsonPathAttributes($path)
     {
         $quote = func_num_args() === 2 ? func_get_arg(1) : "'";
 
-        return (new Collection($path))->map(function ($attribute) {
-            return $this->parseJsonPathArrayKeys($attribute);
-        })->collapse()->map(function ($attribute) use ($quote) {
-            return filter_var($attribute, FILTER_VALIDATE_INT) !== false
-                        ? $attribute
-                        : $quote.$attribute.$quote;
-        })->all();
+        return (new Collection($path))
+            ->map(fn ($attribute) => $this->parseJsonPathArrayKeys($attribute))
```

Comment:
This can be simplified to just
```suggestion
            ->map([ $this, 'parseJsonPathArrayKeys' ])
```

or
```suggestion
            ->map($this->parseJsonPathArrayKeys(...))
```